aosp_diff/preliminary/frameworks/base/23_0023-Limit-maximum-allowed-size-for-a-status-bar-icon.patch

```diff
@@ -0,0 +1,77 @@
+
+Bug: 169255797
+Test: atest StatusBarIconViewTest
+Test: verified that app crashes instead of SysUI following repro steps
+
+Original Author: Beth Thibodeau <ethibodeau@google.com>
+Original Date: Wed, 27 Jan 2021 13:33:34 -0500
+
+Signed-off-by: svenate <salini.venate@intel.com>
+---
+ .../systemui/statusbar/StatusBarIconView.java        | 10 ++++++++++
+ .../systemui/statusbar/StatusBarIconViewTest.java    | 12 +++++++++++-
+ 2 files changed, 21 insertions(+), 1 deletion(-)
+
+diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
+index 8cf8a2299922..25ae5c2dadef 100644
+--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
++++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
+@@ -83,6 +83,9 @@ public class StatusBarIconView extends AnimatedImageView implements StatusIconDi
+     public static final int STATE_DOT = 1;
+     public static final int STATE_HIDDEN = 2;
+ 
++    /** Maximum allowed width or height for an icon drawable */
++    private static final int MAX_IMAGE_SIZE = 500;
++
+     private static final String TAG = "StatusBarIconView";
+     private static final Property<StatusBarIconView, Float> ICON_APPEAR_AMOUNT
+             = new FloatProperty<StatusBarIconView>("iconAppearAmount") {
+@@ -378,6 +381,13 @@ public class StatusBarIconView extends AnimatedImageView implements StatusIconDi
+             Log.w(TAG, "No icon for slot " + mSlot + "; " + mIcon.icon);
+             return false;
+         }
++
++        if (drawable.getIntrinsicWidth() > MAX_IMAGE_SIZE
++                || drawable.getIntrinsicHeight() > MAX_IMAGE_SIZE) {
++            Log.w(TAG, "Drawable is too large " + mIcon);
++            return false;
++        }
++
+         if (withClear) {
+             setImageDrawable(null);
+         }
+diff --git a/packages/SystemUI/tests/src/com/android/systemui/statusbar/StatusBarIconViewTest.java b/packages/SystemUI/tests/src/com/android/systemui/statusbar/StatusBarIconViewTest.java
+index 9971e0cf81a3..a59d8bb0b5b7 100644
+--- a/packages/SystemUI/tests/src/com/android/systemui/statusbar/StatusBarIconViewTest.java
++++ b/packages/SystemUI/tests/src/com/android/systemui/statusbar/StatusBarIconViewTest.java
+@@ -35,6 +35,7 @@ import android.content.ContextWrapper;
+ import android.content.pm.ApplicationInfo;
+ import android.content.pm.PackageManager;
+ import android.content.res.Resources;
++import android.graphics.Bitmap;
+ import android.graphics.Color;
+ import android.graphics.drawable.Icon;
+ import android.os.UserHandle;
+@@ -123,4 +124,13 @@ public class StatusBarIconViewTest extends SysuiTestCase {
+         assertEquals("Transparent backgrounds should fallback to drawable color",
+                 color, mIconView.getStaticDrawableColor());
+     }
+-}
+\ No newline at end of file
```

Comment:
See this. If this is correct.

Reply:
> See this. If this is correct.

Double checked and it looks fine. This was introduced by git format-patch.